File: leetcode/2707-merge-two-2d-arrays-by-summing-values/2707-merge-two-2d-arrays-by-summing-values.cpp
class Solution {
public:
    vector<vector<int>> mergeArrays(vector<vector<int>>& l, vector<vector<int>>& r) {
        map<int, int> tracker;
        for (auto i: l) tracker[i.front()] += i.back();
        for (auto i: r) tracker[i.front()] += i.back();
        vector<vector<int>> res;
        for (auto [key, val]: tracker) res.push_back({key, val});
        return res;
    }
};

File: leetcode/2707-merge-two-2d-arrays-by-summing-values/2707-merge-two-2d-arrays-by-summing-values.rs
use std::collections::BTreeMap;
impl Solution {
    pub fn merge_arrays(l: Vec<Vec<i32>>, r: Vec<Vec<i32>>) -> Vec<Vec<i32>> {
        // let mut tracker = new BtreeMap<i32, i32>;
        // for i in l {
        //     tracker[i[0]] += i[1];
        // }
        // for i in r {
        //     tracker[i[0]] += i[1];
        // }
        // let mut res = new Vec<Vec<i32>>;
        // for 

        let mut tracker = BTreeMap::new();
        for pair in l.iter().chain(r.iter()) {
            if let [key, value] = pair[..] {
                *tracker.entry(key).or_insert(0) += value;
            }
        }
        tracker.into_iter().map(|(k, v)| vec![k, v]).collect()
    }
}